conans/test/unittests/client/generators/premake_test.py

```diff
@@ -89,10 +92,28 @@ def variables_setup_test(self):
         self.assertIn('conan_cppflags_MyPkg1 = {"-fPIE"}', content)
         self.assertIn('conan_cppflags_MyPkg2 = {"-march=native"}', content)
 
-        self.assertIn('conan_sharedlinkflags = {"-framework AudioFoundation", "-framework Cocoa"}', content)
+        self.assertIn('conan_sharedlinkflags = {"-framework AudioFoundation", "-framework Cocoa"}',
+                      content)
         self.assertIn('conan_sharedlinkflags_MyPkg1 = {"-framework Cocoa"}', content)
         self.assertIn('conan_sharedlinkflags_MyPkg2 = {"-framework AudioFoundation"}', content)
 
-        self.assertIn('conan_exelinkflags = {"-framework VideoToolbox", "-framework QuartzCore"}', content)
+        self.assertIn('conan_exelinkflags = {"-framework VideoToolbox", "-framework QuartzCore"}',
+                      content)
         self.assertIn('conan_exelinkflags_MyPkg1 = {"-framework QuartzCore"}', content)
         self.assertIn('conan_exelinkflags_MyPkg2 = {"-framework VideoToolbox"}', content)
+
+    def test_conan_basic_setup_content(self):
+        generator = PremakeGenerator(self.conanfile)
+        content = generator.content
+
+        basic_setup_expected = textwrap.dedent("""
+                function conan_basic_setup()
+                    configurations{conan_build_type}
+                    architecture(conan_arch)
+                    includedirs{conan_includedirs}
+                    libdirs{conan_libdirs}
+                    links{conan_libs}
+                    defines{conan_cppdefines}
+                    bindirs{conan_bindirs}
+                end""")
+        self.assertIn(basic_setup_expected, content)
```

Comment:
I would think about substituting these two test with an equality one, for me it will be easier than checking almost every line of the output. Does it make sense or is it just me that know nothing about premake?

```python
def test_premake_content(self):
    generator = PremakeGenerator(self.conanfile)
    content = generator.content

    self.assertEquals(content, textwrap.dedent("""\
        ....
    """))
```

Reply:
It is just different approaches... I will unify them